src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs

```diff
@@ -60,72 +34,17 @@ public FileUploadsValidatorService(ISubjectService subjectService, IFileTypeServ
         // We cannot rely on the normal upload validation as we want this to be an atomic operation for both files.
         public async Task<Either<ActionResult, bool>> ValidateDataFilesForUpload(Guid releaseId, IFormFile dataFile, IFormFile metaFile, string name)
         {
-            if (string.Equals(dataFile.FileName.ToLower(), metaFile.FileName.ToLower(), OrdinalIgnoreCase))
-            {
-                return ValidationActionResult(DataAndMetadataFilesCannotHaveTheSameName);
-            }
-
-            if (dataFile.Length == 0)
-            {
-                return ValidationActionResult(DataFileCannotBeEmpty);
-            }
-
-            if (metaFile.Length == 0)
-            {
-                return ValidationActionResult(MetadataFileCannotBeEmpty);
-            }
-
-            if (dataFile.FileName.IndexOf(" ", Ordinal) > -1 || 
-                dataFile.FileName.IndexOfAny(Path.GetInvalidFileNameChars()) > -1)
-            {
-                return ValidationActionResult(DataFilenameCannotContainSpacesOrSpecialCharacters);
-            }
-            
-            if (FileContainsSpacesOrSpecialChars(dataFile))
-            {
-                return ValidationActionResult(FilenameCannotContainSpacesOrSpecialCharacters);
-            }
-            
-            if (metaFile.FileName.IndexOf(" ", Ordinal) > -1 || 
-                metaFile.FileName.IndexOfAny(Path.GetInvalidFileNameChars()) > -1)
-            {
-                return ValidationActionResult(MetaFilenameCannotContainSpacesOrSpecialCharacters);
-            }
-
-            if (FileContainsSpacesOrSpecialChars(metaFile))
-            {
-                return ValidationActionResult(FilenameCannotContainSpacesOrSpecialCharacters);
-            }
-
-            if (IsFileExisting(releaseId, ReleaseFileTypes.Data, dataFile.FileName))
-            {
-                return ValidationActionResult(CannotOverwriteDataFile);
-            }
-
-            if (IsFileExisting(releaseId, ReleaseFileTypes.Metadata, metaFile.FileName))
-            {
-                return ValidationActionResult(CannotOverwriteMetadataFile);
-            }
-
-            var dataFilePath = AdminReleasePath(releaseId, ReleaseFileTypes.Data, dataFile.FileName.ToLower());
-            var metadataFilePath = AdminReleasePath(releaseId, ReleaseFileTypes.Data, metaFile.FileName.ToLower());
-
-            if (!IsCsvFile(dataFilePath, dataFile))
-            {
-                return ValidationActionResult(DataFileMustBeCsvFile);
-            }
-
-            if (!IsCsvFile(metadataFilePath, metaFile))
-            {
-                return ValidationActionResult(MetaFileMustBeCsvFile);
-            }
-
-            if (await _subjectService.GetAsync(releaseId, name) != null)
-            {
-                return ValidationActionResult(SubjectTitleMustBeUnique);
-            }
+            return await ValidateDataFileNames(releaseId, dataFile.FileName, metaFile.FileName)
```

Comment:
Let's get unit tests in for all of these validations as they should be pretty easy to write.

Reply:
Have added unit tests just for validating the archive file is valid & invalid for now.